### ISSUE TITLE
A fatal error has been detected by the Java Runtime Environment - SIGSEGV

### ISSUE THREAD

**Issue by jagadeeswar725863** [AdoptOpenJDK/openjdk-support#58]
Hello,

We have a Jama application running  as a docker container, but suddenly application crashed with below error. Could you please help possible root cause.
Memory utilization was fine but load average of the server quite went high.
Platform: Centos 7
Environment: 
JRE version: OpenJDK Runtime Environment (11.0.3+7) (build 11.0.3+7)
Java VM: OpenJDK 64-Bit Server VM (11.0.3+7, mixed mode, tiered, compressed oops, g1 gc, linux-amd64)

A fatal error has been detected by the Java Runtime Environment:

 SIGSEGV (0xb) at pc=0x00007fcf68a140b1, pid=1, tid=97

JRE version: OpenJDK Runtime Environment (11.0.3+7) (build 11.0.3+7)
Java VM: OpenJDK 64-Bit Server VM (11.0.3+7, mixed mode, tiered, compressed oops, g1 gc, linux-amd64)
Problematic frame:
V  [libjvm.so+0xaa30b1]  PhaseIdealLoop::build_loop_late_post(Node*)+0x171

Core dump will be written. Default location: /usr/local/tomcat/bin/core.1

An error report file with more information is saved as:
/usr/local/tomcat/bin/hs_err_pid1.log

Compiler replay data is saved as:
/usr/local/tomcat/bin/replay_pid1.log

If you would like to submit a bug report, please visit:
  https://github.com/AdoptOpenJDK/openjdk-build/issues

error occurred during error reporting (), id 0xb, SIGSEGV (0xb) at pc=0x00007fcf69425a47
error occurred during error reporting (), id 0xb, SIGSEGV (0xb) at pc=0x00007fcf69425a47
error occurred during error reporting (), id 0xb, SIGSEGV (0xb) at pc=0x00007fcf69425a47
======
Thanks

**Comment by grzesuav** [comment]
Looking at problematic frame seems remotely similar to https://bugs.openjdk.java.net/browse/JDK-8181070 which have been fixed long ago in openjdk. @jagadeeswar725863 could you see your error log (/usr/local/tomcat/bin/hs_err_pid1.log) and check for similarities ? If it is regression then openjdk issue should be re-open .

**Comment by jagadeeswar725863** [comment]
@grzesuav Unfortunately those logs were not backed up. I will be taking it next time  if the issue re-occurs again.

**Comment by karianna** [comment]
@jagadeeswar725863 - I'm closing this - it should be fixed in 11.0.6.

**Comment by siim-k** [comment]
Hello. I believe that this issue is not fixed. 

#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007fe42abaa7a1, pid=5537, tid=8801
#
# JRE version: OpenJDK Runtime Environment AdoptOpenJDK (11.0.8+10) (build 11.0.8+10)
# Java VM: OpenJDK 64-Bit Server VM AdoptOpenJDK (11.0.8+10, mixed mode, tiered, compressed oops, g1 gc, linux-amd64)
# Problematic frame:
# V  [libjvm.so+0xab77a1]  PhaseIdealLoop::build_loop_late_post(Node*)+0x171


[hs_err_pid5537.log](https://github.com/AdoptOpenJDK/openjdk-support/files/5041466/hs_err_pid5537.log)

**Comment by aahlenst** [comment]
@karianna What made you think that the bug was fixed with 11.0.6?

https://bugs.openjdk.java.net/browse/JDK-8247481 matches best. Unfortunately, it was closed due to lack of requested feedback.

@siim-k Can you reproduce the problem with https://adoptopenjdk.net/upstream.html?variant=openjdk11&ga=ga? That would be required to convince OpenJDK to reopen the issue.

**Comment by karianna** [comment]
> @karianna What made you think that the bug was fixed with 11.0.6?
> 
> https://bugs.openjdk.java.net/browse/JDK-8247481 matches best. Unfortunately, it was closed due to lack of requested feedback.
> 
> @siim-k Can you reproduce the problem with https://adoptopenjdk.net/upstream.html?variant=openjdk11&ga=ga? That would be required to convince OpenJDK to reopen the issue.

I suspect a mixup on my part - I can't see an 11.0.6 reference for this bug

**Comment by siim-k** [comment]
@aahlenst At the moment this issue has appeared only once in a nightly Jenkins test suite run on EC2 instance. I think this issue needs a rather specific set of circumstances to occur and can't be easily reproduced.

**Comment by aahlenst** [comment]
Will reopen if someone provides steps to reproduce.

**Comment by muldon** [comment]
I am also having this error in version 11.0.9

# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007f36203050ad, pid=5049, tid=5065
#
# JRE version: OpenJDK Runtime Environment (11.0.9+11) (build 11.0.9+11-Ubuntu-0ubuntu1.20.04)
# Java VM: OpenJDK 64-Bit Server VM (11.0.9+11-Ubuntu-0ubuntu1.20.04, mixed mode, sharing, tiered, compressed oops, g1 gc, linux-amd64)
# Problematic frame:
# C  [libjimage.so+0x30ad]

**Comment by aahlenst** [comment]
@muldon That is not the same error, only the same type of error (segmentation fault, but in a different place). If you want help, open a new ticket, provide the full crash dump and code that reproduces the problem. Furthermore, you are running a build created by Ubuntu. Please ensure that the problem happens with a binary downloaded from https://adoptopenjdk.net/upstream.html?variant=openjdk11&ga=ga, too.

**Comment by telcotronics** [comment]
# A fatal error has been detected by the Java Runtime Environment:
#  SIGSEGV (0xb) at pc=0x00007fdde3fdea1a, pid=5390, tid=5440
# JRE version: OpenJDK Runtime Environment (11.0.10+9) (build 11.0.10+9-Ubuntu-0ubuntu1.20.04)
# Java VM: OpenJDK 64-Bit Server VM (11.0.10+9-Ubuntu-0ubuntu1.20.04, mixed mode, tiered, compressed oops, g1 gc, linux-amd64)
# Problematic frame:
# V  [libjvm.so+0x8d3a1a]
# Core dump will be written. Default location: Core dumps may be processed with "/lib/systemd/systemd-coredump %P %u %g %s %t 9223372036854775808 %h" (or dumping to /home/replication-server/SIGMA_LINUX/core.5390)
# An error report file with more information is saved as:
# /home/replication-server/SIGMA_LINUX/hs_err_pid5390.log
# If you would like to submit a bug report, please visit:
#   https://bugs.launchpad.net/ubuntu/+source/openjdk-lts
Abortado (`core' generado)

**Comment by telcotronics** [comment]
error in LINUX Mint

#
# A fatal error has been detected by the Java Runtime Environment:
#
# SIGSEGV (0xb) at pc=0x00007f5c3f381a1a, pid=9089, tid=9141
#
# JRE version: OpenJDK Runtime Environment (11.0.10+9) (build 11.0.10+9-Ubuntu-0ubuntu1.20.04)
# Java VM: OpenJDK 64-Bit Server VM (11.0.10+9-Ubuntu-0ubuntu1.20.04, mixed mode, tiered, compressed oops, g1 gc, linux-amd64)
# Problematic frame:
# V  [libjvm.so+0x8d3a1a]
#
# Core dump will be written. Default location: Core dumps may be processed with "/lib/systemd/systemd-coredump %P %u %g %s %t 9223372036854775808 %h" (or dumping to /home/server-test/NetBeansProjects/facturacion/core.9089)
#
# An error report file with more information is saved as:
# /home/server-test/NetBeansProjects/facturacion/hs_err_pid9089.log
#
# If you would like to submit a bug report, please visit:
#   https://bugs.launchpad.net/ubuntu/+source/openjdk-lts

in ubuntu the app work with JDK8 and jdk12

in Mint no work openJDK8 and jdk11

**Comment by aahlenst** [comment]
@telcotronics It is unclear what you're doing or what the problem encountered by the JVM is (would be in hs_err_pid9089.log). Furthermore, this is a build made by the Ubuntu folks, so we cannot really help.

**Comment by DeepikaSharma5** [comment]
Got the same error. Any one found solution for this?

**Comment by aahlenst** [comment]
@DeepikaSharma5 Show your crash log and code to reproduce.

**Comment by DeepikaSharma5** [comment]
[java_error_in_idea_10696.log](https://github.com/AdoptOpenJDK/openjdk-support/files/6060769/java_error_in_idea_10696.log)
@aahlenst 
When I open Intellij IDEA after few seconds it automatically closing. And then this is the crash log

**Comment by aahlenst** [comment]
@DeepikaSharma5 You do not have the same error. Furthermore, you are using a JDK created by Jetbrains, so we're the wrong place to help. You might have faulty RAM.

**Comment by DeepikaSharma5** [comment]
Oh thankyou so much for your time 🙂

**Comment by telcotronics** [comment]
el error aparece cuando uso  openjfx.
tengo muchas aplicaciones, todas las que NO contenga la tecnologia JAVAFX funciona correcto. este error lo acarreo desde mucho tiempo. 
Cuando Oracle paso el control a Apache.
Se desbarataron todos mi proyectos.
Y he estado corrigiendo fallos que no me dejan centrarme en la productividad por estar corrigiendo errores.

La unica solucion que estoy implementando es usar otro lenguaje de programacion mas estable. y abandonando java.

**Comment by gustawdaniel** [comment]
I noticed this bug on Arch Linux after upgrade of Java by Sdkman.

```
sdk ug
```

shows

```
Upgrade:
java (12.0.2-open < 11.0.11.hs-adpt)
```

After typing `webstorm` I see:

```
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by com.intellij.util.lang.UrlClassLoader (file:/opt/webstorm/lib/util.jar) to field java.lang.ClassLoader.classLoaderValueMap
WARNING: Please consider reporting this to the maintainers of com.intellij.util.lang.UrlClassLoader
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007f78e40140cc, pid=13568, tid=13599
#
# JRE version: OpenJDK Runtime Environment AdoptOpenJDK-11.0.11+9 (11.0.11+9) (build 11.0.11+9)
# Java VM: OpenJDK 64-Bit Server VM AdoptOpenJDK-11.0.11+9 (11.0.11+9, mixed mode, tiered, compressed oops, g1 gc, linux-amd64)
# Problematic frame:
# C  0x00007f78e40140cc
#
# Core dump will be written. Default location: Core dumps may be processed with "/usr/lib/systemd/systemd-coredump %P %u %g %s %t %c %h" (or dumping to /home/builduser/core.13568)
#
# An error report file with more information is saved as:
# /home/builduser/java_error_in_webstorm_13568.log
#
# If you would like to submit a bug report, please visit:
#   https://github.com/AdoptOpenJDK/openjdk-support/issues
# The crash happened outside the Java Virtual Machine in native code.
# See problematic frame for where to report the bug.
#
/bin/webstorm: line 160: 13568 Aborted                 (core dumped) "$JAVA_BIN" -classpath "$CLASSPATH" ${VM_OPTIONS} "-XX:ErrorFile=$HOME/java_error_in_webstorm_%p.log" "-XX:HeapDumpPath=$HOME/java_error_in_webstorm_.hprof" "-Djb.vmOptionsFile=${USER_VM_OPTIONS_FILE:-${VM_OPTIONS_FILE}}" ${IDE_PROPERTIES_PROPERTY} -Djava.system.class.loader=com.intellij.util.lang.PathClassLoader -Didea.vendor.name=JetBrains -Didea.paths.selector=WebStorm2021.2 -Didea.platform.prefix=WebStorm com.intellij.idea.Main "$@"
```

**Comment by gustawdaniel** [comment]
Fortunately I fixed it.

I had `JAVA_HOME` set to `/home/builduser/.sdkman/candidates/java/current`.
I typed

```
export JAVA_HOME=/usr/lib/jvm/java-11-openjdk
```

And now it works displaying some warnings

```
WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by com.intellij.util.lang.UrlClassLoader (file:/opt/webstorm/lib/util.jar) to field java.lang.ClassLoader.classLoaderValueMap
WARNING: Please consider reporting this to the maintainers of com.intellij.util.lang.UrlClassLoader
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
```

**Comment by jason-work** [comment]
I also met this issue when I use opengrok to produce index.
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007fb43def7bd3, pid=23976, tid=0x00007fb426e18700
#
# JRE version: OpenJDK Runtime Environment (8.0_292-b10) (build 1.8.0_292-8u292-b10-0ubuntu1~16.04.1-b10)
# Java VM: OpenJDK 64-Bit Server VM (25.292-b10 mixed mode linux-amd64 compressed oops)
# Problematic frame:
# V  [libjvm.so+0x9cbbd3]

**Comment by jason-work** [comment]
I also met this issue .when I used opengrok to produce index .  Ubuntu 18.04 , open-jdk8 
[hs_err_pid23976.log](https://github.com/adoptium/adoptium-support/files/7159654/hs_err_pid23976.log)

**Comment by telcotronics** [comment]
este error se soluciona incluyendo la librería de openfx.. en el proyecto y los so.. este lo resolví hace mucho tiempo..

**Comment by Loongphy** [comment]
It often happened in ci docker container.
```shell
#
# A fatal error has been detected by the Java Runtime Environment:
#
#  SIGSEGV (0xb) at pc=0x00007ff4318d40f8, pid=9, tid=28
#
# JRE version: OpenJDK Runtime Environment Temurin-17.0.5+8 (17.0.5+8) (build 17.0.5+8)
# Java VM: OpenJDK 64-Bit Server VM Temurin-17.0.5+8 (17.0.5+8, mixed mode, sharing, tiered, compressed oops, compressed class ptrs, g1 gc, linux-amd64)
# Problematic frame:
# v  ~BufferBlob::vtable chunks
#
# Core dump will be written. Default location: /drone/src/core
#
# An error report file with more information is saved as:
# /drone/src/hs_err_pid9.log
Compiled method (c2)   23557 13253       4       com.sun.tools.javac.comp.Attr::check (219 bytes)
 total in heap  [0x00007ff439d56710,0x00007ff439d57d40] = 5680
 relocation     [0x00007ff439d56870,0x00007ff439d56930] = 192
 main code      [0x00007ff439d56940,0x00007ff439d57460] = 2848
 stub code      [0x00007ff439d57460,0x00007ff439d57498] = 56
 oops           [0x00007ff439d57498,0x00007ff439d574d8] = 64
 metadata       [0x00007ff439d574d8,0x00007ff439d575a8] = 208
 scopes data    [0x00007ff439d575a8,0x00007ff439d57948] = 928
 scopes pcs     [0x00007ff439d57948,0x00007ff439d57b98] = 592
 dependencies   [0x00007ff439d57b98,0x00007ff439d57bc0] = 40
 handler table  [0x00007ff439d57bc0,0x00007ff439d57cf8] = 312
 nul chk table  [0x00007ff439d57cf8,0x00007ff439d57d40] = 72
#
# If you would like to submit a bug report, please visit:
#   https://github.com/adoptium/adoptium-support/issues
#
Aborted (core dumped)
```
